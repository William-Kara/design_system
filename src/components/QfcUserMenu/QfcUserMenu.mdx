import { ArgTypes, Meta } from '@storybook/blocks';
import QfcUserMenu from '@/components/QfcUserMenu/QfcUserMenu.vue';

<Meta title="Elements/UserMenu" />

# Definition

Dropdown menu used for the users profil actions.

## Props

The following props are available for the QfcUserMenu component:

<ArgTypes of={QfcUserMenu} />

# Usage

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcUserMenu.scss';
```

# Definition

```html
<template>
  <QfcUserMenu :links="links" avatar="avatar.svg" name="John Doe" />
</template>
```
