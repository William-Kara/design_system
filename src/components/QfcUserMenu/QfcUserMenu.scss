@import '../../styles/bulma/components/dropdown';

.dropdown {
  min-width: 15rem;
  .dropdown-trigger {
    width: 100%;
    .qfc-dropdown-button {
      background-color: $background;
      color: $white;
      border-radius: $br-xs;
      border: $primary;
      width: 100%;
      height: auto;
      padding: 0;
      font-size: $fs-s;
      display: flex;
      justify-content: space-between;

      .qfc-dropdown__name {
        margin-right: auto;
      }
      &:hover {
        background-color: $black-hover;
        color: $white;
      }
      &__icon {
        margin-right: 1rem;
        i {
          font-size: $fs-m-2;
        }
        &__active {
          animation-name: rotate;
          animation-duration: 0.5s;
          @keyframes rotate {
            0% {
              transform: rotate(0deg)
            }
            100% {
              transform: rotate(180deg);
            }
          }
          transform: rotate(180deg);
        }
        &__inactive {
          animation-name: rotateBack;
          animation-duration: 0.5s;
          @keyframes rotateBack {
            0% {
              transform: rotate(180deg)
            }
            100% {
              transform: rotate(0deg);
            }
          }
        }
      }
      .qfc-dropdown-avatar {
        &__holder {
          border-radius: $fs-xl;
          width: $fs-xl;
          height: $fs-xl;
          overflow: hidden;
          margin: 5px 15px;
        }
        &__img {
          width: $fs-xl;
        }
      }
    }
  }
  .qfc-dropdown-menu {
    width: 100%;
    &__content {
      background-color: $background;
      border-radius: $br-xs;
      &__item {
        border-radius: $br-xs;
        padding: $q-generic-space-3;
        font-size: $fs-s;
        &:hover {
          background-color: $black-hover;
          color: $white;
        }
      }
    }
  }
}

