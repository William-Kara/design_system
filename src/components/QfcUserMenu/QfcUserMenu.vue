<template>
  <nav class="dropdown" :class="{ 'is-active': isActive }" @click="toggleActive()">
    <div class="dropdown-trigger">
      <button
        class="button qfc-dropdown-button"
        aria-haspopup="true"
        aria-controls="dropdown-menu3"
      >
        <span class="qfc-dropdown-avatar__holder">
          <img class="qfc-dropdown-avatar__img" :src="avatar" width="45" height="45" alt="avatar" />
        </span>
        <span class="qfc-dropdown__name">{{ name }}</span>
        <span class="qfc-dropdown-button__icon icon is-small">
          <i
            class="material-icons-outlined"
            :class="{
              'qfc-dropdown-button__icon__active': isActive,
              'qfc-dropdown-button__icon__inactive': !isActive,
            }"
          >
            expand_more
          </i>
        </span>
      </button>
    </div>
    <div class="dropdown-menu qfc-dropdown-menu" id="dropdown-menu3" role="menu">
      <div class="dropdown-content qfc-dropdown-menu__content">
        <router-link
          v-for="({ title, link }, index) in links"
          :to="link"
          class="dropdown-item qfc-dropdown-menu__content__item"
          :key="`dropdown-link-${index}`"
          target="_self"
        >
          {{ title }}
        </router-link>
      </div>
    </div>
  </nav>
</template>

<script lang="ts" setup>
import { ref } from 'vue';

const isActive = ref(false);

type Link = {
  title: string;
  link: string;
};
interface Props {
  links: Link[];
  avatar: string;
  name: string;
}
defineProps<Props>();
const toggleActive = () => {
  isActive.value = !isActive.value;
};
</script>
<style lang="scss" scoped src="./QfcUserMenu.scss" />
