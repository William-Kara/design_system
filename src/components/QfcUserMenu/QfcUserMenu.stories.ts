import QfcUserMenu from './QfcUserMenu.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcUserMenu> = {
  component: QfcUserMenu,
  title: 'Elements/UserMenu',
  argTypes: {
    links: {
      defaultValue: [],
      required: true,
    },
    name: { control: 'text', required: true },
    avatar: { control: 'text', required: true },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    links: [
      {
        link: '#',
        title: 'Paramètre du compte',
      },
      {
        link: '#',
        title: 'Formule',
      },
      {
        link: '#',
        title: 'Se déconnecter',
      },
    ],
    name: 'John Doe',
    avatar: '/avatar.svg',
  },
};
