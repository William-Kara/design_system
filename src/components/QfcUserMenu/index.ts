import QfcUserMenu from './QfcUserMenu.vue';

export { QfcUserMenu };
export default QfcUserMenu;
