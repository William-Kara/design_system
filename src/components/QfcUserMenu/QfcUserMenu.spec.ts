import { it, describe, beforeEach, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import QfcUserMenu from './QfcUserMenu.vue';

let wrapper: any = null;

const propsValues = {
  links: [
    {
      link: '#',
      title: 'Paramètre du compte',
    },
    {
      link: '#',
      title: 'Formule',
    },
    {
      link: '#',
      title: 'Se déconnecter',
    },
  ],
  name: 'John Doe',
  avatar: '/avatar.svg',
};

beforeEach(() => {
  wrapper = shallowMount(QfcUserMenu, {
    propsData: propsValues,
  });
});

describe('QfcUserMenu Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
