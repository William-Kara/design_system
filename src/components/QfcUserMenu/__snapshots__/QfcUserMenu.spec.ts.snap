// Vitest Snapshot v1

exports[`QfcUserMenu Component > should render correctly 1`] = `
"<nav data-v-a5e7d66b=\\"\\" class=\\"dropdown\\">
  <div data-v-a5e7d66b=\\"\\" class=\\"dropdown-trigger\\"><button data-v-a5e7d66b=\\"\\" class=\\"button qfc-dropdown-button\\" aria-haspopup=\\"true\\" aria-controls=\\"dropdown-menu3\\"><span data-v-a5e7d66b=\\"\\" class=\\"qfc-dropdown-avatar__holder\\"><img data-v-a5e7d66b=\\"\\" class=\\"qfc-dropdown-avatar__img\\" src=\\"/avatar.svg\\" width=\\"45\\" height=\\"45\\" alt=\\"avatar\\"></span><span data-v-a5e7d66b=\\"\\" class=\\"qfc-dropdown__name\\">John Doe</span><span data-v-a5e7d66b=\\"\\" class=\\"qfc-dropdown-button__icon icon is-small\\"><i data-v-a5e7d66b=\\"\\" class=\\"material-icons-outlined qfc-dropdown-button__icon__inactive\\"> expand_more </i></span></button></div>
  <div data-v-a5e7d66b=\\"\\" class=\\"dropdown-menu qfc-dropdown-menu\\" id=\\"dropdown-menu3\\" role=\\"menu\\">
    <div data-v-a5e7d66b=\\"\\" class=\\"dropdown-content qfc-dropdown-menu__content\\">
      <router-link data-v-a5e7d66b=\\"\\" to=\\"#\\" class=\\"dropdown-item qfc-dropdown-menu__content__item\\" target=\\"_self\\">Paramètre du compte</router-link>
      <router-link data-v-a5e7d66b=\\"\\" to=\\"#\\" class=\\"dropdown-item qfc-dropdown-menu__content__item\\" target=\\"_self\\">Formule</router-link>
      <router-link data-v-a5e7d66b=\\"\\" to=\\"#\\" class=\\"dropdown-item qfc-dropdown-menu__content__item\\" target=\\"_self\\">Se déconnecter</router-link>
    </div>
  </div>
</nav>"
`;
